now we don't try to convert types if certain filters

bcoca opened the pull request:
specially those specifically to return a formated string

willthames commented:
my only concern with this is if people write their own filters, they won't be able to add them to the hardcoded list (I don't have this problem, but we do write our own filters, so I can see how this could happen). Can we make it configurable (with the list as is as default)?

bcoca commented:
good point, will do

willthames commented on lib/ansible/constants.py:
```diff
@@ -261,6 +261,7 @@ def load_config_file():
 
 # characters included in auto-generated passwords
 DEFAULT_PASSWORD_CHARS = ascii_letters + digits + ".,:-_"
+DEFAULT_DONT_TYPE_FILTERS = get_config(p, 'jinja2', 'dont_type_filters', 'ANSIBLE_DONT_TYPE_FILTERS', ['to_json', 'to_nice_json', 'to_yaml', 'ppretty', 'json'], islist=True )
```
The name isn't obvious to me.

How about DEFAULT_AS_IS_FILTERS, 'as_is_filters' and 'ANSIBLE_AS_IS_FILTERS' or similar?

This should be documented too, but I can do that separately if need be.

bcoca committed:
keep string type filters as strings

now we don't try to convert types if using a filter that outputs a specifically formated string
made list of filters configurable

jeremy-audet committed:
Make "make webdocs" compatible with Python 3

The `webdocs` make target fails under Python 3. It fails due to a variety of
syntax errors, such as the use of `except Foo, e` and `print 'foo'`. Fix #13463
by making code compatible with both Python 2 and 3.
